tests/test_client_functional.py

```diff
@@ -2453,6 +2457,40 @@ async def handler_redirect(request):
     assert data == body
 
 
+@pytest.mark.parametrize(
+    "invalid_redirect_url",
+    (
+        # yarl.URL.__new__ raises ValueError
+        "http://:/",
+        "http://example.org:non_int_port/",
+        # # yarl.URL.origin raises ValueError
+        "http:/",
+        "http:/example.com",
+        "http:///example.com",
+        "bluesky://d:i:d",
+        "call:+3801234567",
+        "skype:handle",
+        "slack://instance/room",
+        "steam:code",
+        "twitter://handle",
+    ),
+)
+async def test_invalid_redirect_url(
+    aiohttp_client: Any, invalid_redirect_url: Any
+) -> None:
```

Comment:
With the request @ https://github.com/aio-libs/aiohttp/pull/6722/files#r1472189734, please add test cases for these passed into `client.get()` directly that make sure that the exceptions bubbling up from redirect and direct ones raise `NonHttpURL` in the same manner.

I suppose, it's also useful to make these test cases reusable via a fixture:

```suggestion
@pytest.fixture(
    params=(
        # yarl.URL.__new__ raises ValueError
        "http://:/",
        "http://example.org:non_int_port/",
        # # yarl.URL.origin raises ValueError
        "http:/",
        "http:/example.com",
        "http:///example.com",
        "bluesky://d:i:d",
        "call:+3801234567",
        "skype:handle",
        "slack://instance/room",
        "steam:code",
        "twitter://handle",
    ),
)
def invalid_redirect_url(request: pytest.FixtureRequest) -> str:
    return request.param


async def test_invalid_redirect_url(
    aiohttp_client: Any, invalid_redirect_url: str,
) -> None:
```

Reply:
I will use exception class as parameter so it won't need extraction to fixture, I will resolve this conversation

Reply:
> I will use exception class as parameter so it won't need extraction to fixture, I will resolve this conversation

Without a dedicated test, it's hard to judge whether those exceptions are caused by the initial redirect or now. Maybe, we need several endpoints, redirecting to one another before redirecting to an exceptional URL. Separately, from an endpoint that does such a redirect from the very first response. And separate from an error happening before even attempting an HTTP request.